### ISSUE TITLE
Feature request: play gif file or animation

### ISSUE THREAD

**Issue by Globulle** [pdfpc/pdfpc#332]
### Expected Behaviour
As Pdfpc nicely supports video by using a simple `href` trick, it would be great to also being able to play animated `.gif` files or animations produced with the [LaTeX animate package](https://www.ctan.org/pkg/animate) (currently not displayed properly by Linux pdf viewers).

### Actual Behaviour
A link to a `.gif` file does not produce anything.

#### pdfpc version
pdfpc v4.0.2

#### Used Distribution (GTK version, vala version)
Kubuntu 16.04

**Comment by avillani** [comment]
Hello,
I might add this would be very interesting, not to say awesome, also because it's possible to combine the animate package with TikZ. Currently, there is no viable alternative on Linux to be able to  read pdf produced by animate.
thank you !

**Comment by AndreasBilke** [comment]
Everybody is welcome to submit a PR

**Comment by fnevgeny** [comment]
Re: the `animate` package, I believe it uses javascript to handle embedded animations. I'd say it is not going to work in pdfpc, either: https://bugs.freedesktop.org/show_bug.cgi?id=14433

**Comment by fnevgeny** [comment]
Given the problems mentioned above, I'm inclined to label this issue (technically, these are two separate issues) as `wontfix` and close.

**Comment by cpitclaudel** [comment]
Would it work to use a Gdk.Pixbuf instead of a Gst video for displaying gifs?  I was hoping to prepare a patch, but I've never used Vala/vapi so I'm not sure how to access gdk_pixbuf_animation_new_from_file (https://developer.gnome.org/gdk-pixbuf/stable/gdk-pixbuf-Animations.html)

**Comment by fnevgeny** [comment]
> Would it work to use a Gdk.Pixbuf instead of a Gst video for displaying gifs?

In principle yes, but you'll need to duplicate the whole machinery: playback control (with the widgets), synchronization between the presenter&presentation screens. Currently, all this is done using the gstreamer pipeline. So quite a lot of extra code and maintenance headache. Much easier is to convert an animated GIF to mp4 or like in one second using ffmpeg.

> I was hoping to prepare a patch, but I've never used Vala/vapi so I'm not sure how to access gdk_pixbuf_animation_new_from_file (https://developer.gnome.org/gdk-pixbuf/stable/gdk-pixbuf-Animations.html)

https://valadoc.org/gdk-pixbuf-2.0/Gdk.PixbufAnimation.html etc

**Comment by cpitclaudel** [comment]
> playback control (with the widgets), synchronization between the presenter&presentation screens

I wasn't expecting that there'd be playback control for a gif: it would loop infinitely and autostart.

> https://valadoc.org/gdk-pixbuf-2.0/Gdk.PixbufAnimation.html etc

Thanks!

**Comment by fnevgeny** [comment]
> I wasn't expecting that there'd be playback control for a gif: it would loop infinitely and autostart.

Are you talking about support for animated emoji?!

**Comment by cpitclaudel** [comment]
> Are you talking about support for animated emoji?!

Any gifs, really: how often does a UI let you do playback control on a gif?  I'm thinking of the typical browser UI where gifs autoplay and loop.

**Comment by fnevgeny** [comment]
> I'm thinking of the typical browser UI where gifs autoplay and loop.

One of the most annoying Web "features"... YMMV of course, but I still don't understand why one would want an uncontrollable animation in presentation...

**Comment by cpitclaudel** [comment]
In my latest talk, I have an example of an FPGA running some code to blink a led.  It blinks a message in morse code.  The complete message is a few seconds and I spend about a minute on the slide.  The intent is for it to keep beeping its message for the whole duration of the slide, so having it autostart and loop is what makes sense.

A little further in the talk I demo various ide features with video animations; there, too, autostarting and looping is what I want.

**Comment by fnevgeny** [comment]
Still, why not use one of the supported video formats?

**Comment by cpitclaudel** [comment]
Because of autoplay; the example says that the href feature will be deprecated at some point, so I'm trying not to use that.

**Comment by fnevgeny** [comment]
> Because of autoplay; the example says that the href feature will be deprecated at some point, so I'm trying not to use that.

Well, probably not until autoplay works with `multimedia` :). BTW, autoplay there supports only *single* media per page.

**Comment by fnevgeny** [comment]
BTW, I don't think your PR@poppler is related.  Autoplay in `multimedia` works differently, AFAIR.
